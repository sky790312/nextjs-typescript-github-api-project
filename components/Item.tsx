/* eslint-disable @typescript-eslint/camelcase */
import styled from 'styled-components'
import { GithubRepoSchema } from '@/api/schema'
import Link from 'next/link'

interface Props {
  item: GithubRepoSchema
}

export const Item: React.FC<Props> = ({ item }) => {
  return (
    <ItemContainer>
      <h3>{item.id}</h3>
      <p>
        Name: {item.name} -
        <Link href={item.html_url} passHref={true}>
          <a target='_blank' rel='noreferrer'>
            Link
          </a>
        </Link>
      </p>
    </ItemContainer>
  )
}

const ItemContainer = styled.div`
  padding: 20px;
  cursor: default;
  background-color: ${props => props.theme.colors.gray};

  &:hover {
    opacity: 0.7;
  }
`
