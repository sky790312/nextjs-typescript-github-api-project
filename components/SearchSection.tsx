import React from 'react'
import { TextCenterContainer } from '@/GlobalStyles'

interface Props {
  handleKeyword: (keyword) => void
}

export const SearchSection: React.FC<Props> = React.memo(
  ({ handleKeyword }) => {
    const onKeywordChange = (e: React.ChangeEvent<HTMLInputElement>) =>
      handleKeyword(e.target.value)

    return (
      <TextCenterContainer>
        <h3>search github repo by keyowrd: </h3>
        <input type='text' onChange={onKeywordChange} />
      </TextCenterContainer>
    )
  },
)
