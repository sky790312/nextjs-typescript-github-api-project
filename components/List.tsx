import React from 'react'
import styled from 'styled-components'
import { GithubRepoSchema } from '@/api/schema'
import { Item } from '@/components/Item'

interface Props {
  items: GithubRepoSchema[]
}

export const List: React.FC<Props> = React.memo(({ items }) => {
  return (
    <ListContainer>
      {items?.map(item => (
        <Item key={item.id} item={item} />
      ))}
    </ListContainer>
  )
})

const ListContainer = styled.div`
  position: relative;
  padding: 50px;
`
