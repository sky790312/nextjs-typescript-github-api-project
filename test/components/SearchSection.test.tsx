import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import { SearchSection } from '@/components/SearchSection'

describe('SearchSection component', () => {
  it('matches snapshot', () => {
    const handleKeyword = () => {
      console.warn('test handleKeyword cb function')
    }
    const { asFragment } = render(
      withProviders(<SearchSection handleKeyword={handleKeyword} />),
      {},
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
