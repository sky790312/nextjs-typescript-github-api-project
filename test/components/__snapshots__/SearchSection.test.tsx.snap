// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSection component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GlobalStyles__TextCenterContainer-sc-1m5drgn-1 kwExyG"
  >
    <h3>
      search github repo by keyowrd: 
    </h3>
    <input
      type="text"
    />
  </div>
</DocumentFragment>
`;
